desc/objectives/_generic.py

```diff
@@ -11,12 +12,160 @@
 from desc.compute.utils import _parse_parameterization, get_profiles, get_transforms
 from desc.grid import QuadratureGrid
 from desc.optimizable import OptimizableCollection
-from desc.utils import errorif, parse_argname_change
+from desc.utils import errorif, jaxify, parse_argname_change
 
 from .linear_objectives import _FixedObjective
 from .objective_funs import _Objective, collect_docs
 
 
+class ExternalObjective(_Objective, ABC):
+    """Wrap an external code.
+
+    Similar to ``ObjectiveFromUser``, except derivatives of the objective function are
+    computed with finite differences instead of AD. The function does not need not be
+    JAX transformable.
+
+    The user supplied function must take an Equilibrium or a list of Equilibria as its
```

Comment:
would be good to include a simple example/mockup of what the function should look like as part of the docstring, could be something super simple like we do for `ObjectiveFromUser`

Reply:
should also mention that the function needs to return a single 1d array of floats